server/routes/api/documents/documents.ts

```diff
@@ -343,7 +343,11 @@ router.post(
       const collectionIds = await user.collectionIds();
       where = {
         ...where,
-        collectionId: collectionIds,
+        collectionId: can(user, "readDocument", user.team)
+          ? {
+              [Op.or]: [{ [Op.in]: collectionIds }, { [Op.is]: null }],
+            }
+          : collectionIds,
```

Comment:
Haven't tested it but _theoretically_ it seems this won't include the archived doc which was _individually shared_(the ones that appear under "Shared with me" section) with a user formerly, after the collection is deleted?

Reply:
Great catch - I think this is the behaviour on prod. Probably separate the fix out to another PR so as to not over-complicate this one?

On a related note - I've come across this particular bug in other flows as well (e.g. backlinks, viewed docs). I think it'd be worthwhile to invest a separate effort to implement a solution that handles them all.

Reply:
Yeah, separate PR would be good.